stacks/api/bucket/src/crud.ts

```diff
@@ -0,0 +1,285 @@
+import {Bucket, BucketPreferences, BucketDocument} from "../services";
+import {ObjectId} from "@spica-server/database";
+import {findLocale, hasTranslatedProperties, buildI18nAggregation} from "./locale";
+import * as ACL from "../expression";
+import {
+  createRelationMap,
+  getRelationPipeline,
+  resetNonOverlappingPathsInRelationMap
+} from "./relation";
+import {ForbiddenException} from "@nestjs/common";
+import {getUpdateQueryForPatch} from "./patch";
+
+export async function findDocuments(
+  schema: Bucket,
+  params: {
+    resourceFilter?: object;
+    filter?: object;
+    language?: string;
+    relationPaths: string[][];
+    req: any;
+    sort?: object;
+    skip?: number;
+    limit?: number;
+    projectMap: string[][];
+  },
+  options: {
+    schedule?: boolean;
+    localize?: boolean;
+    paginate?: boolean;
+  },
+  factories: {
+    collection: (id: string | ObjectId) => any;
+    preference: () => Promise<BucketPreferences>;
+    schema: (id: string | ObjectId) => Promise<Bucket>;
+  }
+) {
+  const collection = factories.collection(schema._id);
+
+  const aggregations = [];
+
+  // resourcefilter
+  if (Object.keys(params.resourceFilter || {}).length) {
+    aggregations.push(params.resourceFilter);
+  }
+
+  // scheduled contents
+  aggregations.push({$match: {_schedule: {$exists: !!options.schedule}}});
```

Comment:
```suggestion
  if ( !options.schedule ) {
       aggregations.push({$match: {_schedule: {$exists: false}}});
  }
  
```

Reply:
since we want to keep aggregations as small as possible, this would be more appropriate